api/propagation/propagator.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package propagation
+
+import (
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/tag"
+)
+
+// TextFormatPropagator is an interface that specifies methods to create CarrierInjector
```

Comment:
I'm not sure what this `text format` pertains to. I see text and binary propagators mentioned in [otel spec PR 209](https://github.com/open-telemetry/opentelemetry-specification/pull/209), but without any details. The [spec of propagators](https://github.com/open-telemetry/opentelemetry-specification/blob/c32c8fee0a6488654942c13292d5bd2d7b1fb0ff/specification/api-propagators.md) feels to be specified in a way that this interface seems to satisfy the needs of the binary propagator too? If the "text format" is about the contents of the `tag.Map`, then probably the only thing that needs to be a text are keys and maybe some values. If this is about the way extractors and injectors work (wire format?), I don't see anything limiting them to be textual. So would a name like `Propagator` be fine too?